quinn/src/streams.rs

```diff
@@ -149,6 +151,39 @@ where
         conn.wake();
     }
 
+    /// Completes if/when the peer stops the stream, yielding the error code
+    pub fn stopped(&mut self) -> Stopped<'_, S> {
+        Stopped { stream: self }
+    }
+
+    #[doc(hidden)]
+    pub fn poll_stopped(&mut self, cx: &mut Context) -> Poll<Result<VarInt, WriteError>> {
+        let mut conn = self.conn.lock().unwrap();
+
+        if self.stopped_event.is_none() {
+            match conn.inner.stopped(self.stream) {
+                Ok(Some(error_code)) => return Poll::Ready(Ok(error_code)),
+                Err(_) => return Poll::Ready(Err(WriteError::UnknownStream)),
+                _ => (),
```

Comment:
Let's fill in these wildcards with explicit patterns in the two cases above so we can benefit from rust's totality checking in the future.